src/hssm/utils.py

```diff
@@ -146,6 +153,203 @@ def _get_alias_dict(
+    posterior : xr.Dataset
+        The xarray dataset that contains the draws for
+        all the parameters in the posterior.
+        It must contain the parameters that are needed
+        in the distribution of the response, or
+        the parameters that allow to derive them.
+    kwargs :
+        Parameters that are used to get draws but do
+        not appear in the posterior object or
+        other configuration parameters.
+        For instance, the 'n' in binomial models and multinomial models.
+
+    Returns
+    -------
+    xr.DataArray
+        A data array with the value of the log-likelihood
+        for each chain, draw, and value of the response variable.
+    """
+    # Child classes pass "y_values" through the "y" kwarg
+    y_values = kwargs.pop("y", None)
+
+    # Get the values of the outcome variable
+    if y_values is None:  # when it's not handled by the specific family
+        if data is None:
+            y_values = np.squeeze(model.response_component.term.data)
+        else:
+            y_values = response_evaluate_new_data(model, data)
+
+    response_dist = get_response_dist(model.family)
+    response_term = model.response_component.term
+    kwargs, coords = family._make_dist_kwargs_and_coords(model, posterior, **kwargs)
+
+    # If it's multivariate, it's going to have a fourth coord,
+    # but we actually don't need it. We just need "chain", "draw", "__obs__"
+    coords = dict(list(coords.items())[:3])
+
+    n_chains = len(coords["chain"])
+    n_draws = len(coords["draw"])
+    output_array = np.zeros((n_chains, n_draws, len(y_values)))
+    kwargs_prep = {key_: val_[0][0] for key_, val_ in kwargs.items()}
+    shape_dict = {key_: val_.shape for key_, val_ in kwargs_prep.items()}
+    pt_dict = {
+        key_: (pt.vector(key_, shape=((1,) if val_[0] == 1 else (None,))))
+        for key_, val_ in shape_dict.items()
+    }
+
+    # Compile likelihood function
+    if not response_term.is_constrained:
+        rv_logp = pm.logp(response_dist.dist(**pt_dict), y_values)
+        logp_compiled = pm.compile_pymc(
+            [val_ for key_, val_ in pt_dict.items()], rv_logp
+        )
+    else:
+        # Bounds are scalars, we can safely pick them from the first row
+        lower, upper = response_term.data[0, 1:]
+        lower = lower if lower != -np.inf else None
+        upper = upper if upper != np.inf else None
+
+        # Finally evaluate logp
+        rv_logp = pm.logp(
+            pm.Truncated.dist(
+                response_dist.dist(**kwargs_prep), lower=lower, upper=upper
+            ),
+            y_values,
+        )
+        logp_compiled = pm.compile_pymc(
+            [val_ for key_, val_ in pt_dict.items()], rv_logp
+        )
+
+    # Loop through chain and draws
+    for ids in tqdm(
+        list(itertools.product(coords["chain"].values, coords["draw"].values))
+    ):
+        kwargs_tmp = {
+            key_: (
+                val_[*(ids[0], ids[1]), ...]
```

Comment:
How is this line different from `val_[ids[0], ids[1], ...]`?

Reply:
not differet.